### ISSUE TITLE
Doesn't seem to work with version 12

### ISSUE THREAD

**Issue by goochi1** [claranet/terraform-aws-lambda#46]
Have the following code 
environment {
    variables {
      BUCKET_NAME = "${aws_s3_bucket.s3_bucket.id}"
    }
  }

getting the following error

 22:   environment {

Blocks of type "environment" are not expected here. Did you mean to define
argument "environment"? If so, use the equals sign to assign it a value.

**Comment by goochi1** [comment]
Forked and tried to fix the issues see here https://github.com/goochi1/terraform-aws-lambda

However getting the following error

Error: Unsupported argument

  on .terraform/modules/lambda_ri_reccomendation/lambda.tf line 67, in resource "aws_lambda_function" "lambda_with_dl":
  67:   environment                    = ["${slice( list(var.environment), 0, length(var.environment) == 0 ? 0 : 1 )}"]

An argument named "environment" is not expected here. Did you mean to define a
block of type "environment"?

**Comment by wutianchen** [comment]
@goochi1  I have the same issue, the thing is for resource "aws_lambda_function" now has a "environment block" instread of "environment argument". so it needs to be changed to

```
environment                    {
                                    variables = "${var.environment}"
                                  }
```

I am going to make a PR on this, but do not know if this repo is still actively maintained.

**Comment by lorengordon** [comment]
@wutianchen It is quite active, I'm sure a PR would be appreciated.

**Comment by wutianchen** [comment]
@lorengordon  thanks for the comment. will do that, might add feature like "hcl upgrade", "deploy from s3", "step functions". first needs to go through company open source policy process, might take a week or so

**Comment by raymondbutcher** [comment]
Hi, this will be fixed in #49

**Comment by raymondbutcher** [comment]
This is in version `v1.0.0`
